src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/model/BuildMemory.java

```diff
@@ -406,6 +408,24 @@ public void setEntryFailureMessage(GerritTriggeredEvent event, Run r, String fai
     }
 
     /**
+     * Creates a snapshot clone of the current coordination memory status.
+     *
+     * @return the report
+     */
+    @Nonnull
+    public synchronized BuildMemoryReport report() {
+        BuildMemoryReport report = new BuildMemoryReport();
+        for (Map.Entry<GerritTriggeredEvent, MemoryImprint> entry : memory.entrySet()) {
+            List<Entry> triggered = new LinkedList<Entry>();
```

Comment:
Otoh the article that @svanoort linked to had a big disclaimer in the beginning that the title should be _"Why you should never, ever, EVER use linked-list **for number crunching**"_ Where you need to do some back and forth in the list, and I've never disputed that ;)